In [7]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('ggplot')
#Set the display precision
pd.set_option('precision', 4)

In [8]:
df = pd.DataFrame.from_csv('atus_00007.csv')
df['weighted_TV']=df.BLS_LEIS_TV * df.WT06


In [14]:
#Calculated weighted averages on grouped data
def wavg(group, name):
    d = group['BLS_LEIS_TV']
    w = group[name]
    return (d * w).sum() / w.sum()


In [23]:
#This section groups the data by year as a control
grouped = df.groupby(['YEAR'] )

meanData = grouped.apply(wavg, 'WT06')


In [18]:
#Error calulation using replicate weights, need to use all 160 different sets of data weights
#Find the correct columns
column_names = df.columns.values
indices = [i for i, s in enumerate(column_names) if 'RWT06_' in s]
column_names = column_names[indices]

In [24]:
data = []
for name in column_names:
     data.append((grouped.apply(wavg, name)-meanData)**2)

error = np.sqrt(sum(data)*(4./160))/60


In [49]:
readout = "Mean: %f Error: %s " % (meanData.iloc[0], error.iloc[0])

print(readout)

Mean: 157.394754 Error: 0.029485391123 


These valuses match the values given in the atususersguide on pages 40-41, and are a mean of 157.395, and an error of .0293.